### PR TITLE
`General`: Improve error page

### PR DESCRIPTION
**Pull request by ge65cer**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

Artemis did not show an error page when the requested page does not exist.

### Description

Fixed the handling of 404 errors.
Added two buttons to the error page to navigate to the start page or report an issue in GitHub.

### Steps for Testing

Prerequisites:
- 1 User

1. Navigate to a non-existent URL in the client
2. See the error page

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

### Test Coverage

_unchanged_

### Screenshots

error-page

### CODE REVIEW

File: src/main/webapp/app/shared/layouts/error/error.component.html [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
-            </div>
-            <div [hidden]="!error403" class="alert alert-danger" jhiTranslate="error.http.403">You are not authorized to access this page.</div>
-            <div [hidden]="!error404" class="alert alert-danger" jhiTranslate="error.http.404">The page does not exist.</div>
-        </div>
+        <div [hidden]="!error403 || errorMessage" class="alert alert-danger" jhiTranslate="error.http.403">You are not authorized to access this page.</div>
+        <div [hidden]="!error404 || errorMessage" class="alert alert-danger" jhiTranslate="error.http.404">The page does not exist.</div>
+        <div [hidden]="!errorMessage" class="alert alert-danger">{{ errorMessage }}</div>
+
+        <a class="btn btn-primary" [routerLink]="['/']" jhiTranslate="error.action.home">Home</a>
+        <a class="btn btn-secondary" href="https://github.com/ls1intum/Artemis/issues/new/choose" jhiTranslate="error.action.report">Report bug</a>
```
Comment (Strohgelaender) [inline src/main/webapp/app/shared/layouts/error/error.component.html:10]: The about us component uses https://github.com/ls1intum/Artemis/issues/new?projects=ls1intum/1 which allows the user to just submit any text without using the issue templates. Can you also change the URL there, maybe using a shared constant?

**Review by LiaLism — APPROVED** [review]
The error page works as expected. I did not experience the issues mentioned above. Tested on ts1.

**Review by simon-hng — APPROVED** [review]
Code

**Review by ge65cer — COMMENTED** [review]
There seems to be a routing issue for the communication/messages tab. I need to investigate, maybe these routes are not registered correctly.

**Review by tunargs — APPROVED** [review]
Tested on ts2. works as expected

**Review by Hialus — CHANGES_REQUESTED** [review]
Works as expected for URLs that do not follow any specified routing scheme.
For URLs that follow a specified routing scheme, but are still invalid (e.g. non existent id) the behavior is all over the place.
1. For /courses/{id}: Shows an empty page if the id is invalid
2. For /courses/{id1}/exercises/{id2}: Shows (not redirects) the home screen

Considering that this is likely handled by each page individually this could be done in a follow-up PR, as this would require touching all affected pages. But having consistent behavior would be important IMO. (If you do this also consider noting this down in the docs)

And something major that does not happen in another PR, so it is likely caused by yours: The course management page shows the course overview page: 
grafik
This doesn't even apply only to this page. The general routing for courses seems to be broken. E.g. https://artemis-test1.artemis.in.tum.de/courses/40/exercises/622 also shows the overview page
This needs to be fixed!


Tested on TS10 and verified on TS1, while using TS3 to check the routing on another PR.

**Review by stefanwaldhauser — APPROVED** [review]
Code lgtm!

**Review by Yunqing-Wang — COMMENTED** [review]
image
I got the bad request error message

**Pull request closed without merging**